hil/cli.py

```diff
@@ -797,6 +797,27 @@ def create_admin_user(username, password):
 
 
 @cmd
+def list_active_extensions(ext_type):
+    """List active extensions by type.
+
+    Default types: auth, obm, switches, network_allocators.
+
+    Use the 'all' argument to list all active extensions.
+    """
+    if not config.cfg.has_section('extensions'):
+        sys.exit("There are no active extensions.")
+
+    all_extensions = C.extensions.list_active()
+    if ext_type == 'all':
+        active = [ext[0] for ext in all_extensions]
+        sys.stdout.write('All active extensions: ' + ', '.join(active) + '\n')
+    else:
+        active = [ext[0] for ext in all_extensions if ext_type in ext[0]]
```

Comment:
This is not correct in the general case. There's no reason extension modules have to be at any particular path/name. We have a particular structure to the extensions we ship in-tree, but (a) it's just a convention, and (b) the extension mechanism was designed with third party extensions in mind. An extension could be provided by some other package, and we have no guarantees about the name of the module.

Furthermore, there's not really a way of classifying extensions in general; extensions aren't restricted to simply providing backends for specific mechanisms.